examples/01_standalone_sdk/26_custom_visualizer.py

```diff
@@ -0,0 +1,81 @@
+"""Custom Visualizer Example
+
+This example demonstrates how to create and use a custom visualizer by subclassing
+ConversationVisualizer. This approach provides:
+- Clean, testable code with class-based state management
+- Direct configuration (just pass the visualizer instance to visualizer parameter)
+- Reusable visualizer that can be shared across conversations
+- Better separation of concerns compared to callback functions
+- Event handler registration to avoid long if/elif chains
+
+This demonstrates how you can pass a ConversationVisualizer instance directly
+to the visualizer parameter for clean, reusable visualization logic.
+"""
+
+import logging
+import os
+
+from pydantic import SecretStr
+
+from openhands.sdk import LLM, Conversation
+from openhands.sdk.conversation.visualizer import ConversationVisualizerBase
+from openhands.sdk.event import (
+    Event,
+)
+from openhands.tools.preset.default import get_default_agent
+
+
+class MinimalVisualizer(ConversationVisualizerBase):
+    """A minimal visualizer that print the raw events as they occur."""
+
+    def __init__(self, name: str | None = None):
+        """Initialize the minimal progress visualizer.
+
+        Args:
+            name: Optional name to identify the agent/conversation.
+                                  Note: This simple visualizer doesn't use it in output,
+                                  but accepts it for compatibility with the base class.
+        """
+        # Initialize parent - state will be set later via initialize()
+        super().__init__(name=name)
+
+    def on_event(self, event: Event) -> None:
+        """Handle events for minimal progress visualization."""
+        print(f"\n\n[EVENT] {type(event).__name__}: {event.model_dump_json()[:200]}...")
```

Comment:
NIT: maybe we can include a bit of code on how to access the cost from the events since they get wired in later

Reply:
I guess people can refer to DefaultConversationVisualizer for that if they want..